src/sql/workbench/api/common/extHostModelViewDialog.ts

```diff
@@ -640,31 +660,41 @@ class ModelViewDashboardImpl implements azdata.window.ModelViewDashboard {
 		return this._editor.closeEditor();
 	}
 
-	createTab(tab: azdata.DashboardTab, view: azdata.ModelView): azdata.Tab {
-		if (tab.toolbar) {
+	createTab(dashboardTab: DisposableDashboardTab, view: azdata.ModelView): DisposableTab {
+		let tab: DisposableTab;
+		if (dashboardTab.toolbar) {
 			const flexContainer = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
-			flexContainer.addItem(tab.toolbar, { flex: '0 0 auto' });
-			flexContainer.addItem(tab.content, { flex: '1 1 auto' });
-			return {
-				title: tab.title,
-				id: tab.id,
+			flexContainer.addItem(dashboardTab.toolbar, { flex: '0 0 auto' });
+			flexContainer.addItem(dashboardTab.content, { flex: '1 1 auto' });
+			tab = {
+				title: dashboardTab.title,
+				id: dashboardTab.id,
 				content: flexContainer,
-				icon: tab.icon
+				icon: dashboardTab.icon,
+				dispose: () => {
+					dashboardTab.toolbar.dispose();
```

Comment:
Instead of disposing these separately we should be disposing the container and that should be disposing its child items.

Reply:
The container here dashboardTab is created by extension, which doesn't implement the Dispose API.

Reply:
I'm saying the actual flexContainer that you add the disposable toolbar & content components to (through addItem)

Reply:
There we're registering item for disposal, so we should be good here! I'll clean up :)